go/cmd/vtctldclient/internal/command/reparents.go

```diff
@@ -71,8 +158,118 @@ func commandInitShardPrimary(cmd *cobra.Command, args []string) error {
 
+var plannedReparentShardOptions = struct {
+	NewPrimaryAliasStr   string
+	AvoidPrimaryAliasStr string
+	WaitReplicasTimeout  time.Duration
+}{}
+
+func commandPlannedReparentShard(cmd *cobra.Command, args []string) error {
+	keyspace, shard, err := topoproto.ParseKeyspaceShard(cmd.Flags().Arg(0))
+	if err != nil {
+		return err
+	}
+
+	var (
+		newPrimaryAlias   *topodatapb.TabletAlias
+		avoidPrimaryAlias *topodatapb.TabletAlias
+	)
+
+	if plannedReparentShardOptions.NewPrimaryAliasStr != "" {
+		newPrimaryAlias, err = topoproto.ParseTabletAlias(plannedReparentShardOptions.NewPrimaryAliasStr)
+		if err != nil {
+			return err
+		}
+	}
+
+	if plannedReparentShardOptions.AvoidPrimaryAliasStr != "" {
+		avoidPrimaryAlias, err = topoproto.ParseTabletAlias(plannedReparentShardOptions.AvoidPrimaryAliasStr)
+		if err != nil {
+			return err
+		}
+	}
+
+	cli.FinishedParsing(cmd)
+
+	resp, err := client.PlannedReparentShard(commandCtx, &vtctldatapb.PlannedReparentShardRequest{
+		Keyspace:            keyspace,
+		Shard:               shard,
+		NewPrimary:          newPrimaryAlias,
+		AvoidPrimary:        avoidPrimaryAlias,
+		WaitReplicasTimeout: protoutil.DurationToProto(plannedReparentShardOptions.WaitReplicasTimeout),
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, event := range resp.Events {
+		fmt.Println(logutil.EventString(event))
+	}
+
+	return nil
+}
+
+func commandReparentTablet(cmd *cobra.Command, args []string) error {
+	alias, err := topoproto.ParseTabletAlias(cmd.Flags().Arg(0))
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.ReparentTablet(commandCtx, &vtctldatapb.ReparentTabletRequest{
+		Tablet: alias,
+	})
+	if err != nil {
+		return err
+	}
+
+	data, err := cli.MarshalJSON(resp)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("%s\n", data)
+
+	return nil
+}
+
+func commandTabletExternallyReparented(cmd *cobra.Command, args []string) error {
+	alias, err := topoproto.ParseTabletAlias(cmd.Flags().Arg(0))
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.TabletExternallyReparented(commandCtx, &vtctldatapb.TabletExternallyReparentedRequest{
+		Tablet: alias,
+	})
+	if err != nil {
+		return err
+	}
+
+	data, err := cli.MarshalJSON(resp)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("%s\n", data)
+
+	return nil
+}
+
 func init() {
+	EmergencyReparentShard.Flags().DurationVar(&emergencyReparentShardOptions.WaitReplicasTimeout, "wait-replicas-timeout", *topo.RemoteOperationTimeout, "Time to wait for replicas to catch up in reparenting.")
+	EmergencyReparentShard.Flags().StringVar(&emergencyReparentShardOptions.NewPrimaryAliasStr, "new-primary", "", "Alias of a tablet that should be the new primary. If not specified, the vtctld will select the best candidate to promote.")
+	EmergencyReparentShard.Flags().StringSliceVarP(&emergencyReparentShardOptions.IgnoreReplicaAliasStrList, "ignore-replicas", "i", nil, "Comma-separated, repeated list of replica tablet aliases to ignore during the emergency reparent.")
```

Comment:
```suggestion
	EmergencyReparentShard.Flags().StringSliceVarP(&emergencyReparentShardOptions.IgnoreReplicaAliasStrList, "ignore-replicas", "", nil, "Comma-separated, repeated list of replica tablet aliases to ignore during the emergency reparent.")
```
I assume this is a typo.

Reply:
Why "Comma-separated, repeated list"? Why not just "Comma-separated list"?

Reply:
It's repeated in the sense that you can specify the flag multiple times, so both `-i "${alias1},${alias2}"` _and_ `-i ${alias1} -i ${alias2}` will work (as well as `-i "${alias1},${alias2}" -i ${alias3}`). I'm not sure of a concise way to word that, and what I have here is definitely not straightforward, so I'm very open to suggestions!

Reply:
I get it. Fine for right now, maybe we can revisit later.